src/internal/connector/exchange/service_iterators.go

```diff
@@ -85,12 +73,14 @@ func FilterContainersAndFillCollections(
 
 		cID := *c.GetId()
 
-		// this folder exists (probably), do not delete it.
-		delete(deletedPaths, cID)
-
 		// Only create a collection if the path matches the scope.
 		currPath, ok := pathAndMatch(qp, c, scope)
 		if !ok {
+			if currPath != nil {
+				// this path exists, do not delete it.
+				delete(deletedPaths, currPath.String())
```

Comment:
Sounds to me like we want to keep this block. That most accurately preserves the state of the world (these things exist, and were not modified).  You've only listed the downside to the latter, which makes it seem pretty one-sided.

I think an extension of this question is whether we should treat zero jobs (ie: a container with no changes since the last delta) with the same behavior?  Currently we both skip collections (if the scope doesn't match), and create zero-valued collections (if the scope matches, but doesn't see changes).  It's good to have similar outcomes for both, but we're probably better off to do one or the other.